lib/types.nix

```
@@ -873,8 +873,13 @@ rec {
       description = "function that evaluates to a(n) ${optionDescriptionPhrase (class: class == "noun" || class == "composite") elemType}";
       descriptionClass = "composite";
       check = isFunction;
-      merge = loc: defs:
-        fnArgs: (mergeDefinitions (loc ++ [ "<function body>" ]) elemType (map (fn: { inherit (fn) file; value = fn.value fnArgs; }) defs)).mergedValue;
+      merge = loc: defs: {
+        __functionArgs = lib.pipe defs [
+          (lib.map (fn: lib.functionArgs fn.value))
+          (lib.zipAttrsWith (_: expectedArgs: !lib.any (x: !x) expectedArgs))
```

We could apply De Morgan's law here:

```suggestion
          # An argument attribute has a default when it has a default in all function definitions
          (lib.zipAttrsWith (_: expectedArgs: lib.all (x: x) expectedArgs))
```